src/app/pages/storage/modules/pool-manager/utils/generate-vdevs/grouped-disks.class.ts

```diff
@@ -14,14 +14,19 @@ export class GroupedDisks {
     this.diskMap = getDiskTypeSizeMap(disks);
   }
 
-  findSuitableDisks(category: PoolManagerTopologyCategory): DetailsDisk[] {
+  findSuitableDisks(category: PoolManagerTopologyCategory, usedDisks: DetailsDisk[]): DetailsDisk[] {
```

Comment:
I would double check the logic here. Feel like the bug is somewhere above. For example, there is `removeUsedDisks` in this class. Logic is confusing now.

Reply:
Please see new PR: https://github.com/truenas/webui/pull/10559